### PR TITLE
Classes added, methods added, console app class added

### PR DESCRIPTION
**Pull request by suhanovgleb**

### CODE REVIEW

File: ReflectionTest/ReflectionConsoleTest/Program.cs [hunk 1]
```diff
@@ -1,18 +1,70 @@
...
 using System.Text;
 using System.Threading.Tasks;
+using SomeLibrary;
 
 namespace ReflectionConsoleTest
 {
     class Program
     {
         static void Main(string[] args)
         {
```
Comment (DotNetPart) [inline ReflectionTest/ReflectionConsoleTest/Program.cs]: Modularize the function. Split into separate functions.

File: ReflectionTest/ReflectionConsoleTest/Program.cs [hunk 2]
```diff
@@ -1,18 +1,70 @@
...
+                users.DeleteUser(users.FindByLogin("tsar"));
+                users.DeleteUser(user);
+                users.DeleteUser(user);
+
+                
+                var assemblies = AppDomain.CurrentDomain.GetAssemblies();
+                Type[] typelist = new Type[0];
+                foreach (var assembly in assemblies)
+                {
+                    typelist = assembly.GetTypes().Where(t => t.Namespace == "SomeLibrary").ToArray();
```
Comment (DotNetPart) [inline ReflectionTest/ReflectionConsoleTest/Program.cs:35]: Only last assembly containing SomeLibrary namespace will be processed. Why do you need a filter by SomeLibrary at all?

Comment (DotNetPart) [inline ReflectionTest/ReflectionConsoleTest/Program.cs:35]: Why do you use array here? ToArray just does unnecessary enumeration.

File: ReflectionTest/ReflectionConsoleTest/Program.cs [hunk 3]
```diff
@@ -1,18 +1,70 @@
...
+                }
+
+                if (!typelist.Any())
+                {
+                    throw new Exception("Сan't find the target assembly.");
+                }
+
+                Console.WriteLine("\nReflection test: \n");
+
+                foreach (Type type in typelist)
```
Comment (DotNetPart) [inline ReflectionTest/ReflectionConsoleTest/Program.cs:50]: Why isn't this within assembly enumeration

File: ReflectionTest/ReflectionConsoleTest/Program.cs (hunk 2 above)
Comment (DotNetPart) [inline ReflectionTest/ReflectionConsoleTest/Program.cs:35]: Internal types are missed.

File: ReflectionTest/ReflectionConsoleTest/Program.cs [hunk 4]
```diff
@@ -1,18 +1,70 @@
...
+                {
+                    if (type.IsClass)
+                    {
+                        Console.WriteLine($"--Class name: { type.Name }");
+
+                        foreach (var member in type.GetMembers())
```
Comment (DotNetPart) [inline ReflectionTest/ReflectionConsoleTest/Program.cs:56]: Private/Protected/Static members are missed

File: ReflectionTest/SomeLibrary/User.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SomeLibrary
+{
+    public class User
+    {
+        internal User() { }
```
Comment (DotNetPart) [inline ReflectionTest/SomeLibrary/User.cs:11]: Why do you need a separate internal default constructor?

File: ReflectionTest/SomeLibrary/UserStorage.cs [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+                }
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("Error: " + e.Message);
+            }
+        }
+
+        public User FindByLogin(string login)
+        {
```
Comment (DotNetPart) [inline ReflectionTest/SomeLibrary/UserStorage.cs:44]: return users.Where(...)...

File: ReflectionTest/SomeLibrary/UserStorage.cs [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+            AddUser(UserGenerator.CreateRandomUser());
+        }
+
+        public void DeleteUser(User user)
+        {
+            try
+            {
+                if (users.Remove(user))
+                {
+                    throw new Exception("Can't delete. This user doesn't exist.");
```
Comment (DotNetPart) [inline ReflectionTest/SomeLibrary/UserStorage.cs:34]: Are you sure it's better to throw Exception then return indication that no user is deleted?

File: ReflectionTest/SomeLibrary/UserStorage.cs [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+                }
+            }
+            catch (Exception e)
+            {
```
Comment (DotNetPart) [inline ReflectionTest/SomeLibrary/UserStorage.cs:38]: It's not usual to catch an exception in the same method where it was thrown.
Inaddition it's not thrown further

File: ReflectionTest/SomeLibrary/User.cs (hunk 5 above)
Comment (suhanovgleb) [reply ReflectionTest/SomeLibrary/User.cs:11]: It's used for CreateRandomUser() in UserGenerator class. I made it internal to conform to the principles of encapsulation.

Comment (DotNetPart) [reply ReflectionTest/SomeLibrary/User.cs:11]: This leads to potential issue of not filling Username/Password/IsActive

File: ReflectionTest/SomeLibrary/User.cs [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+        public User(string login, string password, bool isActive = false)
+        {
+            Login = login;
+            Password = password;
+            IsActive = isActive;
+        }
+
+        public string Login { get; set; }
```
Comment (DotNetPart) [inline ReflectionTest/SomeLibrary/User.cs:20]: Set shouldn't be public

File: ReflectionTest/SomeLibrary/User.cs [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+        public string Password { get; set; }
```
Comment (DotNetPart) [inline ReflectionTest/SomeLibrary/User.cs:21]: Same as above

File: ReflectionTest/SomeLibrary/UserStorage.cs (hunk 6 above)
Comment (suhanovgleb) [reply ReflectionTest/SomeLibrary/UserStorage.cs:44]: Replaced: public User FindByLogin(string login) { return users.Where(u => u.Login == login).FirstOrDefault(); }

File: ReflectionTest/SomeLibrary/UserStorage.cs (hunk 8 above)
Comment (suhanovgleb) [reply ReflectionTest/SomeLibrary/UserStorage.cs:38]: Ok, the method no longer throws an exception, so the problem is solved.

File: ReflectionTest/SomeLibrary/UserStorage.cs (hunk 7 above)
Comment (suhanovgleb) [reply ReflectionTest/SomeLibrary/UserStorage.cs:34]: Now this method returns true if the deletion is successful and false, if not.

File: ReflectionTest/SomeLibrary/User.cs (hunk 10 above)
Comment (suhanovgleb) [reply ReflectionTest/SomeLibrary/User.cs:21]: Replaced: public string Password { get; private set; }

File: ReflectionTest/SomeLibrary/User.cs (hunk 9 above)
Comment (suhanovgleb) [reply ReflectionTest/SomeLibrary/User.cs:20]: Replaced: public string Login { get; private set; }

File: ReflectionTest/SomeLibrary/User.cs (hunk 5 above)
Comment (suhanovgleb) [reply ReflectionTest/SomeLibrary/User.cs:11]: Ok, internal default constructor is deleted now.

File: ReflectionTest/ReflectionConsoleTest/Program.cs (hunk 2 above)
Comment (suhanovgleb) [reply ReflectionTest/ReflectionConsoleTest/Program.cs:35]: Should all the assemblies be examined?

Comment (suhanovgleb) [reply ReflectionTest/ReflectionConsoleTest/Program.cs:35]: No longer use ToArray()

Comment (suhanovgleb) [reply ReflectionTest/ReflectionConsoleTest/Program.cs:35]: Now they are displayed.

File: ReflectionTest/ReflectionConsoleTest/Program.cs (hunk 4 above)
Comment (suhanovgleb) [reply ReflectionTest/ReflectionConsoleTest/Program.cs:56]: Now they are displayed.

File: ReflectionTest/ReflectionConsoleTest/Program.cs (hunk 1 above)
Comment (suhanovgleb) [reply ReflectionTest/ReflectionConsoleTest/Program.cs]: Separated work with reflection in the method.

File: ReflectionTest/ReflectionConsoleTest/Program.cs (hunk 3 above)
Comment (suhanovgleb) [reply ReflectionTest/ReflectionConsoleTest/Program.cs:50]: Don't really get the point.

**Pull request closed without merging**